### PR TITLE
publish-commit-bottles: use large GitHub runner

### PR DESCRIPTION
**Pull request by carlocab**
Closes #112134.

### CODE REVIEW

**Comment by Bo98** [comment]
I think any other `github.event.inputs.self_hosted` conditionals can go as we can treat it the same the same as any GitHub-hosted runner.

Might as well rename the input too at that point.

File: .github/workflows/publish-commit-bottles.yml [hunk 1]
```diff
@@ -25,8 +25,7 @@ permissions:
 
 jobs:
   upload:
-    runs-on: ${{github.event.inputs.self_hosted == 'true' && 'linux-self-hosted-1' || 'ubuntu-22.04'}}
-    container: ${{github.event.inputs.self_hosted == 'true' && fromJson('{"image":"ghcr.io/homebrew/ubuntu22.04:master","options":"--user=linuxbrew -e GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED"}') || ''}}
+    runs-on: ${{github.event.inputs.large_runner == 'true' && 'homebrew-large-bottle-upload' || 'ubuntu-22.04'}}
```
Comment (carlocab) [inline .github/workflows/publish-commit-bottles.yml:28]: I think `github.event.inputs.large_runner` will get cast into a boolean anyway, so comparing against `'true'` is probably not needed.

Not sure if this is worth changing here.

**Comment by carlocab** [comment]
> Might as well rename the input too at that point.

Done.

File: .github/workflows/publish-commit-bottles.yml [hunk 2]
```diff
@@ -46,13 +45,13 @@ jobs:
       # Workaround until the `cache` action uses the changes from
       # https://github.com/actions/toolkit/pull/580.
       - name: Unlink workspace
-        if: github.event.inputs.self_hosted != 'true'
+        if: github.event.inputs.large_runner != 'true'
```
Comment (Bo98) [inline .github/workflows/publish-commit-bottles.yml:48]: I mean these can probably go entirely - they exist due to non-ephemeral nature of self-hosted which will no longer be the case.

```suggestion
```

Comment (carlocab) [reply .github/workflows/publish-commit-bottles.yml]: Right, of course

**Review by Bo98 — APPROVED** [review]
Feel free to test this out on a small PR. Ideally something you can fix manually if it goes wrong.

I'll be offline for a few hours but will check back later.

**Comment by carlocab** [comment]
https://github.com/Homebrew/homebrew-core/actions/runs/3167531196

**Comment by carlocab** [comment]
Nope, nothing is happening. I guess it can't find the `homebrew-large-bottle-upload` runner.

**Comment by Bo98** [comment]
Seems to be stuck on "Provisioning", whatever that means.

**Comment by carlocab** [comment]
Is it just that we have to wait in an absurdly long queue in order to use a large runner? If so, doesn't seem worth it.

**Comment by Bo98** [comment]
No it was like that before the run even started.

Probably a billing/account issue or something like that. Will ask about when everyone's back tomorrow.

**Comment by Bo98** [comment]
No, it was like that even before the workflow was run.

I think there could an issue with our account. Will have to wait to tomorrow to ask anyone about it.

**Pull request merged**